### PR TITLE
Add envoy ingress listeners to nerve.conf.json for kubernetes services

### PR DESCRIPTION
**Pull request by analogue**
Main features:
- Supports envoy ingress listeners for kubernetes backed services and adds them to nerve.conf.json accordingly.

Fixed while I was in there:
- broke out envoy stuff into envoy.py
- broke out config data classes into config.py
- replaced ip_address with host_ip or service_ip to disambiguate what the ip addr actually represents (only in new/updated code)

Testing:
- added test coverage with fails under 75%
- verified test coverage execution for newly added code

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.PHONY: help
+help:  ## You're looking at it
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-10s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: test
+test:  ## Run unit tests
+	tox -e xenial
```
Comment (kaisen) [inline src/Makefile]: i don't see a xenial target in tox.ini

File: src/tests/configure_nerve_test.py [hunk 2]
```diff
@@ -126,23 +136,33 @@ def expected_sub_config():
...
+
+    # Convert smartstack mesos services to smartstack k8s services
+    for k, v in expected_sub_config.items():
+        expected_sub_config[k]['host'] = '10.4.5.6'
+
+    new_expected_sub_config = {}
+    for k, v in expected_sub_config.items():
+        new_expected_sub_config[k.replace('10.0.0.1', '10.4.5.6')] = expected_sub_config[k]
+
+    # Add in full mesh envoy configs sor same service
```
Comment (kaisen) [inline src/tests/configure_nerve_test.py]: typo

File: src/nerve_tools/envoy.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            try:
+                envoy_listeners[int(local_host_port)] = \
+                    int(listener['local_address']['socket_address']['port_value'])
+            except KeyError:
+                # If there is no socket_address and port_value, skip this listener
+                pass
+    return envoy_listeners
+
+
+def _get_envoy_service_info(
```
Comment (kaisen) [inline src/nerve_tools/envoy.py]: since we're calling this in configure_nerve.py, let's get rid of the underscore

File: src/Makefile (hunk 1 above)
Comment (analogue) [reply src/Makefile]: This Makefile is for the `/src` directory and refers to the existing [`[testenv:xenial]`](https://github.com/Yelp/nerve-tools/blob/14385148f27b9367aab8806809e543a079bbb731/src/tox.ini#L16)

File: src/tests/configure_nerve_test.py (hunk 2 above)
Comment (analogue) [reply src/tests/configure_nerve_test.py]: Fixed

File: src/nerve_tools/envoy.py (hunk 3 above)
Comment (analogue) [reply src/nerve_tools/envoy.py]: Fixed

File: src/nerve_tools/envoy.py [hunk 4]
```diff
@@ -52,10 +54,26 @@ def get_envoy_ingress_listeners(admin_port: int) -> Mapping[int, int]:
...
+    # to map to the service's envoy ingress port.
+    #
+    # This is not fullproof! multiple instances of the same service may be scheudled
+    # to run on this host. The (service_name, service_ip, service_port) is the only
+    # key that would maintain uniqueness, but service_ip (the pod_id in the case of k8s)
+    # is not part of the information returned from the envoy admin port /listener
+    # response.
+
+    # TODO(spatel) Figure out a tighter solution here
+    key = (service_name, service_info['port'])
```
Comment (analogue) [inline src/nerve_tools/envoy.py]: @kaisen Wanted to make sure I'm not missing something here before I go spend some more time on this.

Comment (kaisen) [reply src/nerve_tools/envoy.py]: this sounds right to me

**Pull request merged**